### PR TITLE
feat: Update gherkin suite to test non-string attributes in fractional

### LINKED ISSUE #378: Add new test suite for non-string attributes in fractional evaluation
### Context
The current [targeting.feature](https://github.com/open-feature/flagd-testbed/blob/main/gherkin/targeting.feature) suit tests the current behaviour of `flagd providers' fractional` where all inputs are coerced to strings before hashing.

As established in [this ADR ](https://github.com/open-feature/flagd/blob/main/docs/architecture-decisions/fractional-non-string-rand-units.md), `fractional` operator will support in the future non-string values as the hashing input. Before implementing this behavior in any language, we should have a suit of tests that guarantees that all SDK providers implement this consistently and correctly. To achieve this, we should add a set of new Gherkin scenarios under a new tag: `@fractional-v3`.

### Scope of Test Scenarios to Add
We will add a new test suite (or update the existing flagd test harness) to cover the following categories:

1. Basic Types & Deterministic Encodings

> Verify that Boolean, String, Integer, Float are correctly encoded and resolved to their expected v3 buckets.

2. CBOR Specific Constraints & Normalizations

> Float-to-Integer Mapping: Conceptually identical values (e.g., 1 and 1.0) must normalize to the same integer type before encoding, resulting in the same hash.

> Zero Values: Both 0.0 and -0.0 must map to the unsigned integer 0.

> Integer Limits: Boundary testing around integer sizes (e.g., 23/24, 255/256 for CBOR width steps) and large integers close to int64 limits to prevent overflow inconsistencies.

> Float Limits: Boundary testing for Float16, Float32, and Float64 maximum/minimum values to ensure correct precision mapping.

3. Complex Types (Objects / Maps)

> Key Ordering: Keys in maps/objects must be sorted lexicographically by their bytes before hashing. A map defined as {"a": 1, "b": 2} must yield the exact same bucket as {"b": 2, "a": 1}.

> Advanced Structures: Test empty maps {}, empty arrays [], nested arrays/maps, and handling of null values inside objects/arrays.

4. Encoding Boundaries & UTF-8 Consistency

> String Lengths: CBOR changes its length descriptor bytes at specific boundaries (23/24 bytes and 255/256 bytes). We must test strings at these exact lengths.

> UTF-8 Multibyte Characters: Validate that emoji sequences, accents, and multibyte characters (e.g., Japanese, special symbols) hash identically across all platforms.

5. Implicit Targeting Key 

> If the hashing input is omitted, flagd must implicitly build a 2-element array [flagKey, targetingKey] and encode it using CBOR.

6. Error & Non-Happy Paths

> Implicit Failures: If the targetingKey is missing or is not a string in implicit mode, the evaluation must fail gracefully, fallback to the default variant, and return an ERROR reason.

> Hardcoded / Evaluated Nulls: Evaluating an explicit null as the hashing input must trigger a PARSE_ERROR or GENERAL error rather than failing silently.

### PR DESCRIPTION
**Pull request by NeaguGeorgiana23**
## This PR
Updates the Gherkin test suite under the a new `@fractional-v3` tag in `targeting.feature` to test the new behavior of the `fractional` operator as defined in the [Fractional Non-String Rand Units ADR](https://github.com/open-feature/flagd/blob/main/docs/architecture-decisions/fractional-non-string-rand-units.md).
Before implementing this behavior across multi-language SDK providers, this test suite guarantees that all SDKs implement CBOR normalization, data structures, and edge-case bucketing consistently and correctly.
Specifically, this PR introduces test scenarios covering:
- **Basic Types & Deterministic Encodings**: Validates Boolean, String, Integer, and Float encodings resolve to consistent, deterministic v3 buckets.
- **CBOR Specific Constraints & Normalizations**:
  - *Float-to-Integer Mapping*: Ensures conceptually identical values (e.g., `1` and `1.0`) normalize to the same integer type before encoding.
  - *Zero Values*: Asserts that both `0.0` and `-0.0` map to the unsigned integer `0`.
  - *Integer & Float Limits*: Validates integer boundary sizes (CBOR width steps at 23/24 and 255/256), large integers near int64 limits, and Float16/32/64 min/max precision boundaries.
- **Complex Types (Objects & Maps)**:
  - *Key Ordering*: Validates that map/object keys are sorted lexicographically by bytes before hashing (e.g., `{"a": 1, "b": 2}` yields the exact same bucket as `{"b": 2, "a": 1}`).
  - *Advanced Structures*: Validates empty maps `{}`, empty arrays `[]`, nested collections, and null value handling inside structures.
- **Encoding Boundaries & UTF-8 Consistency**:
  - *String Lengths*: Tests string encoding at exact CBOR length descriptor thresholds (23/24 bytes and 255/256 bytes).
  - *UTF-8 Multibyte Characters*: Validates cross-platform hashing consistency for emoji sequences, accents, and multibyte symbols (e.g., Japanese characters).
- **Implicit Targeting Key**: Validates that when hashing input is omitted, flagd implicitly builds and CBOR-encodes a 2-element array: `[flagKey, targetingKey]`.
- **Error & Non-Happy Paths**:
  - *Implicit Failures*: Asserts that missing or non-string `targetingKey` values in implicit mode fail gracefully, falling back to the default variant with a `DEFAULT` reason.
  - *Explicit Nulls*: Validates that explicit nulls as hashing input trigger  a silent fallback to the `defaultVariant`.

### Related Issues
Fixes #378 

### Notes
- **Non-Breaking for Existing SDKs**: The new scenarios are isolated under the `@fractional-v3` tag. Existing SDK implementations (currently running `@fractional-v2` or `@fractional-v1`) will not fail in CI as long as their test runners filter out `v3` during the transition period.

### Follow-up Tasks
- [ ] Provider SDK teams (Go, Java, JS, Python, C#, etc.) to implement CBOR-based fractional bucketing and enable `@fractional-v3` in their respective test suites.

### CODE REVIEW

File: gherkin/targeting.feature [hunk 1]
```diff
@@ -428,3 +460,231 @@ Feature: Targeting rules
...
+  Scenario Outline: Valid well-formed UTF-8 String consistency across different languages
+    Given a String-flag with key "fractional-basic-flag" and a default value "fallback"
+    And a context containing a key "hashing_input", with type "String" and with value "<hashing_input>"
+    When the flag was evaluated with details
+    Then the resolved details value should be "<value>"
+
+    Examples: v3
+      | hashing_input           | value    |
+      | あいうえおかき          | bucket17 |
+      | あいうえおかきく        | bucket16 |
```
Comment (toddbaert) [inline gherkin/targeting.feature]: nit: spacing issue here

File: gherkin/targeting.feature [hunk 2]
```diff
@@ -428,3 +460,231 @@ Feature: Targeting rules
...
+      | ééééééééééé             | bucket16 |
+      | éééééééééééé            | bucket19 |
+      | café façade résumé café | bucket9  |
+      | A\u0000B                | bucket15 |
+      | e\u0301                 | bucket7  |
+      | \uD83D\uDE00            | bucket23 |
```
Comment (toddbaert) [inline gherkin/targeting.feature:663]: Just checking intent on these last three rows; they look like escape/UTF encodings, so I want to confirm they're meant to be hashed literally rather than decoded first. I have a draft impl from earlier and the buckets are correct as written, so no blocker; just making sure this isn't an oversight.

File: gherkin/targeting.feature [hunk 3]
```diff
@@ -359,6 +390,7 @@ Feature: Targeting rules
       | missing-variant-targeting-flag    | 3     | GENERAL     |
       | non-string-variant-targeting-flag | 2     |             |
       | empty-targeting-flag              | 1     |             |
+      | targeting-null-flag               | 2     |             |
```
Comment (toddbaert) [inline gherkin/targeting.feature:393]: The PR description says explicit nulls should "trigger a PARSE_ERROR or GENERAL error rather than failing silently," but this row asserts an empty error code, i.e. a silent fallback to defaultVariant (value 2). Those contradict each other.

I guessing this is just because, as we discussed, we decided the new error behavior should be opt in? Again, I just want to make sure we are on the same page.

**Review by toddbaert — APPROVED** [review]
I left a couple things for you to confirm, and I will leave one more here: 

We mentioned wanting an "opt-in" new error handling paradigm, wherein targeting errors would cause erroneous evaluations; however, that's not implemented here (in fact the opposite is explicitly asserted). I'm fine with that, I just want to mention it. in case it was an oversight. Maybe you intend to work on that separately (no problem).

File: gherkin/targeting.feature [hunk 4]
```diff
@@ -428,3 +463,243 @@ Feature: Targeting rules
...
+    | {"x": [{"a": {"b": []}}] }                              | bucket16 |
+    | {"x": [{"a": {"b": null}}] }                            | bucket6  |
+    | {"x": [{"a": {"b": {}}}] }                              | bucket2  |
+    | {"": "", "": [], "": {}, "": null}                      | bucket25 |
+    | {"": {"": ""}, "": ["", {"": ""}], "": {"": {"": ""}}}  | bucket11 | 
+    | {"a": [[[[[]]]]] }                                      | bucket15 |  
+    | {"a": [[[[null]]]] }                                    | bucket7  |
+    | {"a": {"b": {"c": {"d": {}}} } }                        | bucket21 | 
+    | {"a": {"b": {"c": {"d": null}} } }                      | bucket19 | 
+
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply gherkin/targeting.feature]: ^ interesting point - maybe something we want to explicitly test so we should keep it in? Could be problematic though, we'll see, I guess.

File: gherkin/targeting.feature (hunk 3 above)
Comment (NeaguGeorgiana23) [reply gherkin/targeting.feature:393]: Yes, the tests scenarios for now are meant to test the default behavior, the one that falls back. The tests for the opt in behavior will come in a future PR. 

I will update the PR description since that part is not correct.

**Comment by NeaguGeorgiana23** [comment]
> I left a couple things for you to confirm, and I will leave one more here:
> 
> We mentioned wanting an "opt-in" new error handling paradigm, wherein targeting errors would cause erroneous evaluations; however, that's not implemented here (in fact the opposite is explicitly asserted). I'm fine with that, I just want to mention it. in case it was an oversight. Maybe you intend to work on that separately (no problem).

Yes, this PR only deals with our default behaviour and a future PR will create tests for this opt-in strict mode.

File: gherkin/targeting.feature (hunk 2 above)
Comment (NeaguGeorgiana23) [reply gherkin/targeting.feature:663]: Yes, the intent is for these to be hashed literally as written ("A\u0000B", "e\u0301", "\uD83D\uDE00").
